From fdde18c3416afd1e971ddee7a65f08bed9fd44a7 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 19 Jul 2021 10:47:02 +0100
Subject: [PATCH] ARM: dts: bcm2711: Tidy the HDMI I2C aliases

The bcm2711 vc3-kms-v3d overlay enables the I2C instances used for
EDID data. Give these distinct I2C interface numbers (20 & 21) to
clearly separate them from other regular I2C blocks (1, 3-6) and the
mux on I2C0 (10+).

The 2711 DTS tree no longer includes i2c2, so the explicit deletion can
be removed.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2711-rpi-4-b.dts | 10 ++--------
 arch/arm/boot/dts/bcm2711-rpi-400.dts | 10 ++--------
 arch/arm/boot/dts/bcm2711-rpi-cm4.dts | 10 ++--------
 3 files changed, 6 insertions(+), 24 deletions(-)

--- a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
@@ -350,11 +350,12 @@
 		mmc0 = &emmc2;
 		mmc1 = &mmcnr;
 		mmc2 = &sdhost;
-		/delete-property/ i2c2;
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
 		i2c5 = &i2c5;
 		i2c6 = &i2c6;
+		i2c20 = &ddc0;
+		i2c21 = &ddc1;
 		spi3 = &spi3;
 		spi4 = &spi4;
 		spi5 = &spi5;
@@ -560,13 +561,6 @@
 	pinctrl-0 = <&i2s_pins>;
 };
 
-/ {
-	__overrides__ {
-		/delete-property/ i2c2_baudrate;
-		/delete-property/ i2c2_iknowwhatimdoing;
-	};
-};
-
 // =============================================
 // Board specific stuff here
 
--- a/arch/arm/boot/dts/bcm2711-rpi-400.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-400.dts
@@ -348,11 +348,12 @@
 		mmc0 = &emmc2;
 		mmc1 = &mmcnr;
 		mmc2 = &sdhost;
-		/delete-property/ i2c2;
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
 		i2c5 = &i2c5;
 		i2c6 = &i2c6;
+		i2c20 = &ddc0;
+		i2c21 = &ddc1;
 		spi3 = &spi3;
 		spi4 = &spi4;
 		spi5 = &spi5;
@@ -558,13 +559,6 @@
 	pinctrl-0 = <&i2s_pins>;
 };
 
-/ {
-	__overrides__ {
-		/delete-property/ i2c2_baudrate;
-		/delete-property/ i2c2_iknowwhatimdoing;
-	};
-};
-
 // =============================================
 // Board specific stuff here
 
--- a/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
@@ -363,11 +363,12 @@
 		mmc0 = &emmc2;
 		mmc1 = &mmcnr;
 		mmc2 = &sdhost;
-		/delete-property/ i2c2;
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
 		i2c5 = &i2c5;
 		i2c6 = &i2c6;
+		i2c20 = &ddc0;
+		i2c21 = &ddc1;
 		spi3 = &spi3;
 		spi4 = &spi4;
 		spi5 = &spi5;
@@ -573,13 +574,6 @@
 	pinctrl-0 = <&i2s_pins>;
 };
 
-/ {
-	__overrides__ {
-		/delete-property/ i2c2_baudrate;
-		/delete-property/ i2c2_iknowwhatimdoing;
-	};
-};
-
 // =============================================
 // Board specific stuff here
 
